org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/LookupEnvironment.java

```diff
@@ -154,6 +154,7 @@ static enum CompleteTypeBindingsSteps {
 		NONE,
 		CHECK_AND_SET_IMPORTS,
 		CONNECT_TYPE_HIERARCHY,
+		SEAL_TYPE_HIERARCHY,
```

Comment:
I'm happy to see this structure extended by @srikanth-sankaran -- it seems to be helpful after all :)